threedi_schema/application/schema.py

```diff
@@ -113,6 +113,19 @@ def upgrade(
         Specify 'convert_to_geopackage=True' to also convert from spatialite
         to geopackage file version after the upgrade.
         """
+        try:
+            rev_nr = get_schema_version() if revision == "head" else int(revision)
+        except ValueError:
+            raise ValueError(
+                f"Incorrect version format: {revision}. Expected 'head' or a numeric value."
+            )
+        if convert_to_geopackage and rev_nr < 300:
+            warnings.warn(
+                "Cannot convert to geopackage for revision {=revision.str} because geopackage support is "
```

Comment:
The formatting is off here
```suggestion
                f"Cannot convert to geopackage for revision {revision} because geopackage support is "
```

Also you should let it error here instead of doing something different than the user requested.

Reply:
Fixed in 652cb4c2631d62688021b2316c30e0a35d7f5315